hydro_cli/src/core/gcp.rs

```diff
@@ -60,25 +60,39 @@ impl LaunchedBinary for LaunchedComputeEngineBinary {
     }
 }
 
-struct LaunchedComputeEngine {
+pub struct LaunchedComputeEngine {
     resource_result: Arc<ResourceResult>,
-    external_ip: String,
+    pub internal_ip: String,
+    pub external_ip: Option<String>,
     binary_counter: RwLock<usize>,
 }
 
 #[async_trait]
 impl LaunchedHost for LaunchedComputeEngine {
+    fn get_bind_config(&self, bind_type: &BindType) -> BindConfig {
+        match bind_type {
+            BindType::UnixSocket => BindConfig::UnixSocket,
+            BindType::InternalTcpPort => BindConfig::TcpPort(self.internal_ip.clone()),
+            BindType::ExternalTcpPort(_) => todo!(),
+        }
+    }
+
     async fn launch_binary(&self, binary: &Path) -> Result<Arc<RwLock<dyn LaunchedBinary>>> {
+        let target_addr = SocketAddr::new(
+            self.external_ip
```

Comment:
I'm not sure if we need to require external IPs for binary launches.

Reply:
Hmm we need it here to SSH into the machine, but the assumption that an external IP is necessary I guess is not correct when the machine being used to perform the deploy is also in the same VPC (but this currently can never be the case because we create a new VPC for the deployment).

I guess a proper solution could be to proxy through a bastion.

Reply:
#418

Reply:
Oh I see; I forgot that my "runner" bastion is still not in the same VPC. Would it be easier to add an option to use an existing vpc? I'm a little worried (potentially for no reason) about costs if all our VMs need external IPs.

Reply:
I think we should be fine on costs, since we are only charged for egress. So as long as we are not dumping a bunch of logs streamed over the SSH connections it should be very minimal. It is nice to support a bastion though, so will leave that on the backlog.

Using an existing VPC will need a bit more thinking, since it's easy to get confusing deployments if there are existing firewall rules that the CLI is not aware of.